src/providers/ldap/ldap_id.c

```diff
@@ -292,7 +297,13 @@ struct tevent_req *users_get_send(TALLOC_CTX *memctx,
         }
     }
 
-    if (state->use_id_mapping || filter_type == BE_FILTER_SECID) {
+    if (state->non_posix == true) {
+        state->filter = talloc_asprintf(state,
```

Comment:
(btw as long as we agree that `if (val)` and `if (!val)` is used for bools only then I think it's OK readability wise but I still prefer `ptr != NULL` over `!ptr`